avocado/core/test.py

```diff
@@ -583,6 +590,9 @@ def _run_avocado(self):
         except exceptions.TestDecoratorSkip as details:
             stacktrace.log_exc_info(sys.exc_info(), logger='avocado.test')
             raise exceptions.TestSkipError(details)
+        except exceptions.TestCancel as details:
+            stacktrace.log_exc_info(sys.exc_info(), logger='avocado.test')
+            raise exceptions.TestCancel(details)
```

Comment:
Another way to write this is simply:

```python
raise
````

Which, IMHO, makes it clearer that you're raising the same thing you received.  But it's a matter of preference, so pick your own style.